jack-sys/src/functions.rs

```diff
@@ -0,0 +1,3983 @@
+            jackctl_parameter_get_default_value_impl,
+            jackctl_parameter_has_range_constraint_impl,
+            jackctl_parameter_has_enum_constraint_impl,
+            jackctl_parameter_get_enum_constraints_count_impl,
+            jackctl_parameter_get_enum_constraint_value_impl,
+            jackctl_parameter_get_enum_constraint_description_impl,
+            jackctl_parameter_get_range_constraint_impl,
+            jackctl_parameter_constraint_is_strict_impl,
+            jackctl_parameter_constraint_is_fake_value_impl,
+            jack_error_impl,
+            jack_info_impl,
+            jack_log_impl,
+            jack_set_property_impl,
+            jack_get_property_impl,
+            jack_free_description_impl,
+            jack_get_properties_impl,
+            jack_get_all_properties_impl,
+            jack_remove_property_impl,
+            jack_remove_properties_impl,
+            jack_remove_all_properties_impl,
+            jack_set_property_change_callback_impl,
+            jack_get_internal_client_name_impl,
+            jack_internal_client_handle_impl,
+            jack_internal_client_load_impl,
+            jack_internal_client_unload_impl,
+            jack_get_max_delayed_usecs_impl,
+            jack_get_xrun_delayed_usecs_impl,
+            jack_reset_max_delayed_usecs_impl,
+            jack_midi_get_event_count_impl,
+            jack_midi_event_get_impl,
+            jack_midi_clear_buffer_impl,
+            jack_midi_reset_buffer_impl,
+            jack_midi_max_event_size_impl,
+            jack_midi_event_reserve_impl,
+            jack_midi_event_write_impl,
+            jack_midi_get_lost_event_count_impl,
+            jack_ringbuffer_create_impl,
+            jack_ringbuffer_free_impl,
+            jack_ringbuffer_get_read_vector_impl,
+            jack_ringbuffer_get_write_vector_impl,
+            jack_ringbuffer_read_impl,
+            jack_ringbuffer_peek_impl,
+            jack_ringbuffer_read_advance_impl,
+            jack_ringbuffer_read_space_impl,
+            jack_ringbuffer_mlock_impl,
+            jack_ringbuffer_reset_impl,
+            jack_ringbuffer_reset_size_impl,
+            jack_ringbuffer_write_impl,
+            jack_ringbuffer_write_advance_impl,
+            jack_ringbuffer_write_space_impl,
+            jack_uuid_to_index_impl,
+            jack_uuid_compare_impl,
+            jack_uuid_copy_impl,
+            jack_uuid_clear_impl,
+            jack_uuid_parse_impl,
+            jack_uuid_unparse_impl,
+            jack_uuid_empty_impl,
+        }
+    };
+}
+
+pub unsafe fn jack_release_timebase(client: *mut jack_client_t) -> ::libc::c_int {
+    let f = LIB.jack_release_timebase_impl;
+    f(client)
+}
+pub unsafe fn jack_get_cycle_times(
+    client: *const jack_client_t,
+    current_frames: *mut jack_nframes_t,
+    current_usecs: *mut jack_time_t,
+    next_usecs: *mut jack_time_t,
+    period_usecs: *mut ::libc::c_float,
+) -> ::libc::c_int {
```

Comment:
weak functions should probably return `Option<T>` instead of panic to allow clients to recover.